x-pack/plugins/lens/public/visualizations/xy/visualization_helpers.tsx

```diff
@@ -406,20 +407,24 @@ export function validateLayersForDimension(
   // Multiple layers must be consistent:
   // * either a dimension is missing in ALL of them
   // * or should not miss on any
-  if (layers.every(missingCriteria) || !layers.some(missingCriteria)) {
+  if (dataLayers.every(missingCriteria) || !dataLayers.some(missingCriteria)) {
     return { valid: true };
   }
   // otherwise it's an error and it has to be reported
-  const layerMissingAccessors = layers.reduce((missing: number[], layer, i) => {
-    if (missingCriteria(layer)) {
+  const layerMissingAccessors = allLayers.reduce((missing: number[], layer, i) => {
```

Comment:
following my previous comment, I think we could do  here
```suggestion
  const layerMissingAccessors = dataLayers.reduce((missing: number[], layer, i) => {
```

Reply:
aaaah, I see you're taking the index here, ok, then nevermind